java/gateway-core/src/main/java/com/avaulta/gateway/rules/MultiTypeBulkDataRules.java

```diff
@@ -0,0 +1,36 @@
+package com.avaulta.gateway.rules;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.*;
+import lombok.extern.java.Log;
+
+import java.util.Map;
+
+@Builder(toBuilder = true)
+@Log
+@AllArgsConstructor //for builder
+@NoArgsConstructor //for Jackson
+@Getter
+@EqualsAndHashCode
+@JsonInclude(JsonInclude.Include.NON_NULL) //NOTE: despite name, also affects YAML encoding
+public class MultiTypeBulkDataRules implements RuleSet {
+
+    /**
+     * map of file path templates to rules for matching files
+     *
+     * eg, /export/{week}/data_{shard}.csv -> CsvRules
```

Comment:
I'd just be explicit with the names -> 
`ColumnarRules` -> `CSVRules`
`RecordRules` -> `NDJSONRules`

Actually both are `RecordRules` of different record types.

Reply:
yeah. Have gotten rid of CSVRules in favor of just ColumnarRules; can have `\t` as delimiter in theory to support TSV if we care.

But then yes - RecordRules could support CSV/TSV format and provide equivalent functionality. It's just not as simple in the simple case.  We could move to it from ColumnarRules in the future major version though if we want to consolidate.